src/tools/fuzzing.h

```diff
@@ -2997,12 +3018,27 @@ class TranslateToFuzzReader {
 
   Type getEqReferenceType() { return pick(getEqReferenceTypes()); }
 
+  Type getMVPType() {
+    return pick(items(FeatureOptions<Type>().add(
+      FeatureSet::MVP, Type::i32, Type::i64, Type::f32, Type::f64)));
+  }
+
   Type getTupleType() {
     std::vector<Type> elements;
-    size_t numElements = 2 + upTo(MAX_TUPLE_SIZE - 1);
-    elements.resize(numElements);
-    for (size_t i = 0; i < numElements; ++i) {
-      elements[i] = getSingleConcreteType();
+    size_t maxElements = 2 + upTo(MAX_TUPLE_SIZE - 1);
+    for (size_t i = 0; i < maxElements; ++i) {
+      auto type = getSingleConcreteType();
+      // Don't add ref types into a tuple for now, as if they are non-nullable
+      // then they can't be spilled into locals. And even if they are nullable,
+      // like a funcref, but we create a subtype that is *not* nullable (like
+      // a ref.func) then that would be the effective type here, and again we'd
+      // have a non-nullable type.
```

Comment:
I recommend disallowing `funcref` everywhere because it is non-nullable and enforcing the use of `(ref null func)` instead.

Reply:
I was wrong when I wrote that `funcref` is non-nullable, and I think our offline discussions have resolved the disconnect here. Still, could you simplify this code by just having `getSingleConcreteType` return only nullable reference types? I don't understand the issue you describe here:

>  And even if they are nullable, like a funcref, but we create a subtype that is *not* nullable (like a ref.func) then that would be the effective type here, and again we'd have a non-nullable type.